### PR TITLE
ST fix round7

### PR DESCRIPTION
**Pull request by glevkovich**
- RVBManager: add a detection mechanism -pruning is done during checkpoint
- RVBManager: add initial metrics support
- RangeValidationTree: rename addNode and removeNode
- RangeValidationTree: move inline from cpp to header
- RangeValidationTree: remove kNodeValueMax_
- test_skvbc_persistence: enable back 2 tests
- RangeValidationTree: printToLog add label, uncomment DO_DEBUG+cosmetics
- BCStateTran, RVBManager: validate span on empty data received
- st-ctl: additional fixes
- State Transfer: return when parent digest not found, print debug info
- RVBManager: fix bug during updateRvbDataDuringCheckpoint
- RVBManager: fix bug in ST source
- RVBManager: fix bug during pruning reports
- RVT_test: Change the test verifying basic math properties
- BCStateTran: Add StartCollectingState Handler,refactor  event handlers
- BCStateTran: minor fixes

### CODE REVIEW

**Review by mdarade — APPROVED** [review]
Looks good.
Thanks Gil.

**Review by StefanPalashev — APPROVED** [review]
LGTM, thanks @glevkovich.

File: bftengine/src/bcstatetransfer/BCStateTran.cpp [hunk 1]
```diff
@@ -199,6 +199,20 @@ BCStateTran::Metrics BCStateTran::createRegisterMetrics() {
       metrics_component_.RegisterStatus("current_rvb_data_state", "")};
 }
 
+void BCStateTran::bindEventsHandlers() {
+  if (config_.runInSeparateThread) {
+    incomingEventsQ_ = std::make_unique<concord::util::Handoff>(config_.myReplicaId, "incomingEventsQ");
+    incomingStateTransferMsgHandler_ = std::bind(&BCStateTran::handleIncomingStateTransferMessage, this, _1, _2, _3);
```
Comment (arc-vmware) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:205]: Just adding a test comment, will delete in a min,

Comment (arc-vmware) [inline bftengine/src/bcstatetransfer/BCStateTran.cpp:205]: this is a dummy comment

**Pull request merged**